buildSrc/src/main/groovy/com/google/firebase/gradle/plugins/measurement/apksize/GenerateVariantPlugin.groovy

```diff
@@ -0,0 +1,47 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+package com.google.firebase.gradle.plugins.measurement.apksize
+
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+
+class GenerateVariantPlugin implements Plugin<Project> {
+
+    @Override
+    void apply(Project project) {
+        def childProjects = project.rootProject.childProjects
+        def firebaseProjectsNames = childProjects.findAll { isFirebaseProject(it.value) }.keySet()
+        project.android {
+            firebaseProjectsNames.each {
+                productFlavors.create("$it") {
+                    dimension "apkSize"
+                }
+            }
+            firebaseProjectsNames.each {
+                sourceSets.getByName(it) { sourceSet ->
+                    sourceSet.java.srcDirs += ['src/empty/java']
+                }
+            }
```

Comment:
Is this needed? Empty doesn't contain anything of value and can probably be deleted as well.

Reply:
Empty is the base "flavor" we use to measure all the others. I like the idea of having it as a separate flavor to make this clear. I include it's source code in all the others in case we would need to make a change in it down the road. But it's not strictly necessary. I'd remove it if you feel strongly about it.

Reply:
I agree that we should have it as a separate flavor, but it should be completely empty. Now that we've eliminated the samples, it doesn't need the SampleCode class. At this point, there's no need to include the source in the other flavors, because it should always be empty.

If you want to keep the source directory in each flavor to prevent future errors, perhaps we should rename it to `base` or something like that. That would better describe its purpose anyway.

Reply:
I've removed the source dependency and `SampleCode`, and I've renamed it to base as it's a much better name to what it actually is. Thanks!